perf_test/blas/blas/KokkosBlas_team_dot_perf_test.hpp

```diff
@@ -0,0 +1,132 @@
+  Kokkos::View<Scalar*, MemSpace> x;
+
+  // Create 1D view w/ Device as the ExecSpace; this is the output vector
+  Kokkos::View<Scalar*, MemSpace> y;
+
+  testData_rps_team_dot(int m):m(m) {
+
+
+  x = Kokkos::View<Scalar* , MemSpace>(Kokkos::ViewAllocateWithoutInitializing("x"), m);
+  y = Kokkos::View<Scalar* , MemSpace>(Kokkos::ViewAllocateWithoutInitializing("y"), m);
+
+  Kokkos::deep_copy(x, 3.0);
+  Kokkos::deep_copy(y, 2.0);
+
+  }
+};
+
+/* Taking in by reference avoids making a copy of the data in memory, whereas
+ * taking in by value would make a copy in memory.  Copying operations do not
+ * enhance performance.
+ *  
+ *  A function takes data as a pointer when you're dealing with a collection of
+ *  things, such as 8 test datasets
+ *
+ */
+// Declaring the machinery needed to run as an RPS test
+
+#ifdef KOKKOSKERNELS_ENABLE_TESTS_AND_PERFSUITE
+test_list construct_team_dot_kernel_base(const rajaperf::RunParams& params);
+#endif
+
+// Templated function 
+template<typename ExecSpace, typename Layout>
+testData_rps_team_dot<ExecSpace, Layout> setup_test(int m,
+                                       int repeat,
+                                       bool layoutLeft,
+                                       const int numberOfTeams
+                                       );
+
+                                
+
+// Must have the full function body, as templated functions are recipes for
+// functions
+//
+template <class ExecSpace, class Layout>
+void run(int m, int repeat) {
+
+  std::cout << "Running BLAS Level 1 TEAMPOLICY-BASED DOT performance experiment ("
+            << ExecSpace::name() << ")\n";
+
+  std::cout << "Each test input vector has a length of " << m << std::endl;
+
+  // Declaring variable pool w/ a seeded random number;
+  // a parallel random number generator, so you
+  // won't get the same number with a given seed each time
+
+  // Constructing an instance of testData, which takes m as a param
+  testData_rps_team_dot<ExecSpace,Layout> testMatrices(m);
+
+
+
+  // do a warm up run of team dot:
+  Kokkos::parallel_for("TeamDotUsage_RPS_Warm_Up",
+                       policy(testMatrices.numberOfTeams, Kokkos::AUTO),
+                       KOKKOS_LAMBDA(const typename testData_rps_team_dot<ExecSpace,Layout>::member_type& team) {
+                       KokkosBlas::Experimental::dot(team, testMatrices.x, testMatrices.y);
+                       });
+
+
+  // The livepo/git_submodules_rps_PR
+  Kokkos::fence();
+  Kokkos::Timer timer;
+
+    for (int i = 0; i < testMatrices.repeat; i++) {
+           
+  Kokkos::parallel_for("TeamDotUsage_RPS",
+                       policy(testMatrices.numberOfTeams, Kokkos::AUTO),
+                       KOKKOS_LAMBDA(const typename testData_rps_team_dot<ExecSpace,Layout>::member_type& team) {
+                       double result = KokkosBlas::Experimental::dot(team, testMatrices.x, testMatrices.y);
+                       });
+
+    ExecSpace().fence();
```

Comment:
@brian-kelley - timer.seconds() occurs in a commented bloc in this file.  I will delete this comment, because it does confuse things.  Performance results are output to the "RAJAPerf-timing.csv" and other files (to you point about recording the data somewhere).